src/ROOTNTupleReader.cc

```diff
@@ -0,0 +1,212 @@
+
+  auto collectionType = m_metadata_readers[filename]->GetView<std::vector<std::string>>(root_utils::collInfoName(category));
+  m_collectionInfo[category].type = collectionType(0);
+   
+  auto subsetCollection = m_metadata_readers[filename]->GetView<std::vector<short>>(root_utils::subsetCollection(category));
+  m_collectionInfo[category].isSubsetCollection = subsetCollection(0);
+
+  return true;
+}
+
+void ROOTNTupleReader::openFile(const std::string& filename) {
+  openFiles({filename});
+}
+
+void ROOTNTupleReader::openFiles(const std::vector<std::string>& filenames) {
+
+  m_filenames.insert(m_filenames.end(), filenames.begin(), filenames.end());
+  for (auto& filename : filenames) {
+    if (m_metadata_readers.find(filename) == m_metadata_readers.end()) {
+      m_metadata_readers[filename] = ROOT::Experimental::RNTupleReader::Open(root_utils::metaTreeName, filename);
+    }
+  }
+
+  m_metadata = ROOT::Experimental::RNTupleReader::Open(root_utils::metaTreeName, filenames[0]);
+
+  auto versionView = m_metadata->GetView<std::vector<uint16_t>>(root_utils::versionBranchName);
+  auto version = versionView(0);
+
+  m_fileVersion = podio::version::Version{version[0], version[1], version[2]};
+
+  auto edmView = m_metadata->GetView<std::vector<std::tuple<std::string, std::string>>>(root_utils::edmDefBranchName);
+  auto edm = edmView(0);
+
+  auto availableCategoriesField = m_metadata->GetView<std::vector<std::string>>(root_utils::availableCategories);
+  m_availableCategories = availableCategoriesField(0);
+
+}
+
+unsigned ROOTNTupleReader::getEntries(const std::string& name) {
+  if (m_readers.find(name) == m_readers.end()) {
+    for (auto& filename : m_filenames) {
+      try {
+        m_readers[name].emplace_back(ROOT::Experimental::RNTupleReader::Open(name, filename));
+      }
+      catch (const ROOT::Experimental::RException& e) {
+        std::cout << "Category " << name << " not found in file " << filename << std::endl;
+      }
+    }
+  }
+  m_totalEntries[name] = std::accumulate(m_readers[name].begin(), m_readers[name].end(), 0, [](int total, auto& reader) {return total + reader->GetNEntries();});
+  return m_totalEntries[name];
+}
+
+std::unique_ptr<ROOTFrameData> ROOTNTupleReader::readNextEntry(const std::string& name) {
+  return readEntry(name, m_entries[name]);
+}
+
+std::unique_ptr<ROOTFrameData> ROOTNTupleReader::readEntry(const std::string& category, const unsigned entNum) {
+  if (m_totalEntries.find(category) == m_totalEntries.end()) {
+    getEntries(category);
+  }
+  if (entNum >= m_totalEntries[category]) {
+    return nullptr;
+  }
+
+  if (m_collectionInfo.find(category) == m_collectionInfo.end()) {
+    if (!initCategory(category)) {
+      return nullptr;
+    }
+  }
+
+  m_entries[category] = entNum+1;
+
+  ROOTFrameData::BufferMap buffers;
+  auto dentry = m_readers[category][0]->GetModel()->GetDefaultEntry();
+
+  std::map<std::pair<std::string, int>, std::vector<podio::ObjectID>*> tmp;
```

Comment:
If I understand correctly here: This map of `vector<ObjectID>*` is necessary, because we cannot directly "connect" the `buffers.references[i].get()` to the `vector*` that is expected in `CaptureValueUnsafe`?

If that is the case, could you put a small comment here stating why we need this map?

Reply:
This is related to `buffers.references[i].get()` being a `unique_ptr` I think. I'll double check but I think passing either the `unique_ptr` or the `unique_ptr.get()` both crash and that's why I had to choose that not very elegant solution since I don't know better :shrug:

Reply:
Hmm, pity. Would have been nice if the `get()` version would have worked. But if both crash, then this is the way it has to be for the moment.